core/src/main/scala/org/apache/spark/shuffle/FileShuffleBlockResolver.scala

```diff
@@ -104,11 +106,12 @@ private[spark] class FileShuffleBlockResolver(conf: SparkConf)
    * Get a ShuffleWriterGroup for the given map task, which will register it as complete
    * when the writers are closed successfully
    */
-  def forMapTask(shuffleId: Int, mapId: Int, numBuckets: Int, serializer: Serializer,
-      writeMetrics: ShuffleWriteMetrics): ShuffleWriterGroup = {
+  def forMapTask(shuffleId: Int, mapId: Int, stageAttemptId: Int, numBuckets: Int,
+      serializer: Serializer, writeMetrics: ShuffleWriteMetrics): ShuffleWriterGroup = {
```

Comment:
I made the change in this case.  I took a look at a few other cases, but didn't really feel like it would help.  Eg., in `IndexShuffleBlockResolver`, there are a handful of methods I added a `stageAttemptId` to.  I could have instead grouped it with the existing `shuffleId`, but is that grouping really so much more important than the other params to those methods, eg. the `mapId`?  I tried a bit of refactoring and that grouping felt awkward to me.